libnetwork/network.go

```diff
@@ -1411,24 +1411,20 @@ func (n *network) addSvcRecords(eID, name, serviceID string, epIP, epIPv6 net.IP
 
 	sr, ok := c.svcRecords[networkID]
```

Comment:
Unrelated to your PR; do you know why DNS entries are stored in the controller instead of the network? I'm wondering if things could be made the other way around, as the current implementation requires the controller to be locked each and every time a DNS entry is updated or a DNS query is handled. This looks wrong from my point of view.

Reply:
It also looks wrong to me. The (predecessor to) `svcRecords` was moved from the network to the controller in https://github.com/moby/libnetwork/commit/25f496f771e66f333ec8153471091fd7f31ffb3e so presumably it had something to do with libnetwork's persistence model and the (now obsolete) reliance on a distributed KV store for propagating state to nodes.